File: qwe/mapping/CV_Map.c
/*
 ===========================================================================================
 Name        : CV_Map.c
 Author      : Ricker
 Version     :
 Copyright   :
 Description :

 Map of playing field with simple interface to individually check tiles.

 I have increased the width and length of the map to map the actual distances of the ramps
 on to the 2-D map so that no angle calculations are required to calculate the distances of
 the ramps during real time.  To compensate, widths of the walls have been adjusted to ensure
 that the playing field dimensions are preserved (walls along length and width of ground level
 do not have the same thickness). Alternatively, the flat distances could be mapped and the
 actual distance of the ramp could be calculated from the angle during real time.

 There has been a problem of using to much memory for the map array in this program.  By changing
 the data types in variables in TileProp struct from int to short and Bool, the map is now able
 to handle 8 tiles per inch. The goal was to have 16 tiles per inch to not lose any resolution from the
 actual course map (If 16 is used the way it is now seg fault occurs).  Since this is
 the static map, I have commented out the struct elements color and status to try to
 preserve memory but this was not enough to make 16 tiles per inch possible. It may be that 8
 tiles per inch is good enough.  Suggestions for improving the program to increase resolution
 where to use malloc, some system of pointers, or code the map directly in CV.

 Note that there still needs to be some way to visualize the code in this program to ensure that
 it has been coded properly.
 ============================================================================================
 */

#include <stdio.h>
#include <stdlib.h>
//#include <math.h>

int main(){

	struct TileProp
	{
		short type;	//driving surface (0), start (1), storage (2), land (3), sea (4), air(5), marker/white lines (7), wall (8)
		//_Bool status; //empty (0), filled (1)	// if status and color are uncommented the check tile interface will have to be fixed
		//short color; //unknown(0) blue(1), black(2), green(3), yellow (4), red(5), brown (6), white(7)
		short level; //ground (0), ramp (1), lower platform (2), upper platform (3)
		_Bool path; // not path (0) path (1)  //this element was suggested for implementing a path for the bot to stay on
	};

	//double cmPerIn = 2.54;
	int tileSize = 8; // tiles per in
	int width = 97.4375*tileSize;	// width in tiles course width (in. * tiles per in)
									//width is the distance from the edge of upper platform to the edge of lower platform
									// 24 + 24 + 49.4375 = 97.4375
	int length = 73.75*tileSize;  	// length in tiles course length (in. * tiles per in)
									//length is the distance from the edge of upper platform to edge of ground level where start area is


	struct TileProp map[length][width];	//make map of tiles

	//loop variables
	int i;
	int j;
	int k;
	int m;
	//variables for tile check
	char ans1 = 'y';
	int row = 0;
	int col = 0;

	int upPltW = 24*tileSize;			//Upper platform width
	int upPltL = 24*tileSize;			//Upper platform length
	int upRmpW = 49.4375*tileSize;		//upper ramp width
	int upRmpL = 24*tileSize;			//upper ramp length
	int loPltW = 24*tileSize;			//lower platform width
	int loPltL = 24*tileSize;			//lower platform length
	int loRmpW = 24*tileSize;			//lower ramp width
	int loRmpL = 24.75*tileSize;		//lower ramp length
	int fatWall = 1.125 *tileSize;		// the thicker wall, on the east and west sides of course
	int skinnyWall = .98675*tileSize;	// the thinner wall, on north and south sides of course
	int whiteLine = 0.5 * tileSize;		//thickness of white lines
	int startW = 12 * tileSize;			//start square width
	int startL = 12 * tileSize;			//start square length


	for (i = 0; i < length; i++)  //general initialization for entire board
	{
		for (j = 0; j < width; j++)
		{
			//map[i][j].status = 0; //define all areas as empty
			map[i][j].path = 0;	//define entire area as not path
			//map[i][j].color = 2; //define as black
			map[i][j].type = 0; //define all area as driving surface
			map[i][j].level = 0; //define all area as ground
		}
	}

	for (i = 0; i < upPltL; i++) // define upper platform
	{
			for (j = 0; j < upPltW; j++)
			{
				map[i][j].level = 3; //define area as upper platform
			}
	}

	for (i = 0; i < upRmpL; i++) //define upper ramp

		for (j = upPltW - 1; j < (width - loPltW - 1); j++)
		{
			map[i][j].level = 1; //define area as ramp
		}

	for (i = 0; i < loPltL; i++) // define lower platform
	{
			for (j = width - 1 - loPltW; j < width; j++)
			{
				map[i][j].level = 2;  //define as lower platform
			}
	}

	for (i = loPltL - 1; i < (loPltL + loRmpL); i++) //define lower ramp
	{
		for (j = (width - 1 - loRmpW); j < width; j++)
		{
			map[i][j].level = 1;  //define as ramp
		}
	}

	//walls
	for (i = (length - skinnyWall - 1); i < length; i++)	//define long wall along width of course (south side)
	{
		for (j = 0; j < width; j++)
		{
			map[i][j].type = 8;	//define as wall
		}
	}

	for (i = upPltL - 1; i < upPltL + skinnyWall; i++)	//define short wall along width of course (north side)
	{
		for (j = 0; j < upPltW + upRmpW; j++)
		{
			map[i][j].type = 8;	//define as wall
		}
	}

	for (i = upPltL - 1; i < length; i++)	//define long wall along length of course (west side)
	{
		for (j = 0; j < fatWall; j++)
		{
			map[i][j].type = 8;	//define as wall
		}
	}

	for (i = loPltL + loRmpL; i < length; i++)	//define short wall along length of course (east side)
	{
		for (j = width - fatWall - 1; j < width; j++)
		{
			map[i][j].type = 8;	//define as wall
		}
	}

	// start area
	for (i = length - 1 - skinnyWall - startL - whiteLine ; i < length - skinnyWall; i++)  	//define white outline of start area
	{																						// this loop includes the actual start area
		for (j = fatWall; j < fatWall + startW + whiteLine; j++)							// next loop will fix the enclosed start area
		{
			map[i][j].type = 7; //define as marker/white lines
			//map[i][j].color = 7; //define as white
		}
	}

	for (i = length - 1 - skinnyWall - startL; i < length - skinnyWall; i++)	// fix enclosed start area
	{
		for (j = fatWall; j < fatWall + startW; j++)
		{
			map[i][j].type = 1; //define as start
			//map[i][j].color = 2;  //define as black
		}
	}

	//air loading zone
	for (i = 8.75*tileSize - 1; i < (24 - 8.75)*tileSize; i++) 	//define white outline of air storage
										// this loop includes actual air dropoff area, next loop will fix the dropoff area
	{									// loop after that will take care of white line separating each indiv storage area
		for(j = 0; j < 3.5*tileSize; j++)
		{
			map[i][j].type = 7; //define as marker/white lines
			//map[i][j].color = 7; //define as white
		}
	}

	for (i = (8.75+0.5)*tileSize -1; i < (24-8.75-0.5)*tileSize; i++) 	//fix enclosed air storage space
	{
		for(j = 0; j < 3*tileSize; j++)
		{
			map[i][j].type = 5;	//define as air
			//map[i][j].color = 0; //define as unknown color
		}
	}

	for(i = (8.75+0.5+2.5)*tileSize - 1; i < 0.5*tileSize; i ++) //make separating line between adjacent air storage spaces
	{
		for (j = 0; j < 3.5*tileSize; j++)
		{
			map[i][j].type = 7; //define as marker/white lines
			//map[i][j].color = 7; //define as white
		}
	}

// storage area
	for (i = upPltL + skinnyWall - 1; i < upPltL + skinnyWall + 6.5*tileSize; i++)	//define white outline of cargo storage
	{																		// this loop includes actual storage area, next loop will fix the storage area
																			// loop after that will take care of white line separating each indiv storage area
		for (j = fatWall+ 14.875*tileSize -1; j < fatWall + (14.875+ 42.5)*tileSize; j++)
		{
			map[i][j].type = 7; //define as marker/white lines
			//map[i][j].color = 7; //define as white
		}
	}

	k = 0;
	m = fatWall + (14.875+3)*tileSize;
	while(k<=12)					// fix enclosed storage area
	{
		m = m + 3*tileSize*k;
		for(i = upPltL + skinnyWall -1; i < upPltL + skinnyWall + 6*tileSize; i++)
		{
			for (j = m-1 ; j < m + 2.5*tileSize; j++)
			{
				map[i][j].type = 2;		//type is storage
				//map[i][j].color = 0;	//color is unknown
			}
		}
		k = k + 1;
	}
	/*
	for (i = upPltL + skinnyWall - 1; i < upPltW + skinnyWall + 6*tileSize; i++)	//fix enclosed storage area
	{
		for (j = fatWall+(14.875+0.5)*tileSize - 1; j < (14.875+14*0.5+14*2.5)*tileSize; j++)
		{
			map[i][j].type = 2;	//define as storage
			//map[i][j].color = 0; //define as unknown color
		}
	}

	j = fatWall + (14.875+0.5+2.5)*tileSize - 1;			//make separating white lines in storage area
	k = fatWall + (14.875+14*0.5+14*2.5)*tileSize;
	for (i = upPltL + skinnyWall - 1; i < upPltL + skinnyWall + 6.5*tileSize; i++)
	{
		while (j < k)
		{
			map[i][j].type = 7;	//define as marker
			//map[i][j].color = 7;	//define as white color
			j = j + 3*tileSize;
		}
	}
	*/

	//sea loading zone
	for (i = upPltL+skinnyWall+8.25*tileSize-1; i < length-skinnyWall-(12.5+8.25)*tileSize; i++) 	//define white outline of sea storage
										// this loop includes actual sea dropoff area, next loop will fix the dropoff area
	{
		for(j = 0; j < 4.5*tileSize; j++)
		{
			map[i][j].type = 7; //define as marker/white lines
			//map[i][j].color = 7; //define as white
		}
	}

	k = 0;
	m = upPltL + skinnyWall + (8.25+3)*tileSize;
	while(k <= 4)
	{
		for (i = m-1; i < m + 2.5*tileSize; i++);
		{
			for(j = fatWall -1; j < fatWall + 4*tileSize; j++)
				map[i][j].type = 4; //define as sea
				//map[i][j].color = 0; //define as unknown
		}
		k = k + 1;
	}


	/*
	for (i = upPltL+skinnyWall+8.75*tileSize; i < length-skinnyWall-(12.5+8.75)*tileSize; i++) 	//fix enclosed air storage space
	{
		for(j = 0; j < 4*tileSize; j++)
		{
			map[i][j].type = 4;	//define as sea
			//map[i][j].color = 0; //define as unknown color
		}
	}

	i = upPltL+skinnyWall+(8.25+3)*tileSize - 1;			//make separating white lines in storage area
	k = length - skinnyWall - (12.5+8.25+.5)*tileSize;
	while (i < k)
	{
		for (j = fatWall - 1; j < fatWall + 4.5*tileSize; j++)
		{
			//map[i][j].color = 7;
			map[i][j].type = 7;
		}
		i = i + 3*tileSize;
	}
	*/

	//land loading zone
	for (i = length-skinnyWall-5.5*tileSize - 1; i < length - skinnyWall; i++)	//define white outline of land storage
	{																		// this loop includes actual land area, next loop will fix the storage area

		for (j = fatWall+(12.5+32.25)*tileSize-1; j < width - fatWall - 32.25*tileSize; j++)
		{
			map[i][j].type = 7; //define as marker/white lines
			//map[i][j].color = 7; //define as white
		}
	}

	k = 0;
	m = fatWall + (12.5+32.25+3)*tileSize;
	while(k<=4)		// fix enclosed land storage area
	{
		m = m + 3*tileSize*k;
		for(i = length - fatWall - 5*tileSize; i < length - fatWall; i++)
		{
			for (j = m-1 ; j < m + 2.5*tileSize; j++)
			{
				map[i][j].type = 3;		//type is land
				//map[i][j].color = 0;	//color is unknown
			}
		}
		k = k + 1;
	}

	// short checking loop, still need some sort of visualization to confirm map
	printf("Want to check the properties of a tile? (y/n)\n");
	scanf("%c", &ans1);
	while(ans1 != 'n')
	{
		printf("fyi map[0][0] is the corner of the upper most platform");
		printf("enter the row number as a positive int.\n");
		scanf("%d", &row);
		printf("enter the column number as a positive int. \n");
		scanf("%d", &col);
		if ((row < 0) || (col < 0))
		{
			printf("please enter positive values");
			continue;
		}
		if ((row < length) && (col < width))
		{
				printf("map[row][col].type = %d.\n"
				//"map[row][col].status = %d.\n"
				//"map[row][col].color = %d.\n"
				"map[row][col].level = %d.\n"
				"map[row][col].path = %d.\n", map[row][col].type, /*map[row][col].status, map[row][col].color,*/ map[row][col].level, map[row][col].path);
		}
		else
		{
			printf("the coordinate does not exist.\n");
		}
		printf("Would you like to check another tile? (y/n)\n");
		scanf("%c", &ans1);
	}
	printf("goodbye\n");


	return 0;
}

File: qwe/mapping/CV_Map.py
"""
CV Map 
Ricker Snow

The mk_map function generates the map as it appears on sheet 7 of the course schematic; the element in the first column and first row corresponds to the corner closest to the upper platform.  mk_map accepts 'res' argument defining the resolution of the map in tiles per inch.  mk_map returns the map.


Notes:  Will require angle calculations for ramp distance calculations.  There may be some attributes we want to remove, like path and maybe status.  There's also redundancy between desc and color properties in that color contains white and desc contains markers/white lines.  This code is not optimized as it will not be used during bot runtime.  This code makes the map which will be stored as a pickled file or a text file.  The pickled file or text file will be brought back in to memory at bot runtime. 
"""
import numpy as np

def mk_map(res, map_grid_vars, map_prop_vars):

	class MapClass:
		def __init__(self, res, mapSize):
			self.res = res	
			self.mapSize = mapSize	
			self.scale = res
			# make a numpy 2D array
			self.grid = np.zeros(mapSize, dtype = [('desc', np.uint8),('status', np.uint8),('color', np.uint8),('level', np.uint8),('path', np.uint8)])
		def ydim(self): 	#return the y dimension of grid (number of rows)
			return(len(self.grid))
		def xdim(self):		#return the x dimension of grid (number of columns)
			return(len(self.grid[0]))	
		def fill(self, p1, p2, prop):	#fill an area defined by p1 and p2 with value prop['key'] in layer key of MyMap.grid
			x1,y1,x2,y2 = p1[0],p1[1],p2[0],p2[1]
			if y1 <= y2:	y_range = (y1, y2)	
			else: y_range = (y2, y1)
			if x1 <= x2: x_range = (x1, x2)
			else: x_range = (x2, x1)
			for y in xrange(y_range[0],y_range[1]):	
				for x in xrange(x_range[0],y_range[1]):
					for key in prop:
						self.grid[y][x][key] = prop[key]

	mapSize = (map_grid_vars['height'], map_grid_vars['width']) # (height, width) --> (rows, cols)	
	myMap = MapClass(res,mapSize)


	"""
	Begin initializing board, everything is first initialized to:
	empty, not path, black, driving surface, and ground level.
	"""
	for y in xrange(0,map_grid_vars['height']):	
		for x in xrange(0,map_grid_vars['width']):
			myMap.grid[y][x]['level'] = map_prop_vars['ground']
			myMap.grid[y][x]['desc'] = map_prop_vars['driv_srfc']
			myMap.grid[y][x]['color'] = map_prop_vars['black']
			myMap.grid[y][x]['path'] = map_prop_vars['not_path']
			myMap.grid[y][x]['status'] = map_prop_vars['empty']
	#define upper platform
	for y in xrange(0,map_grid_vars['upPltH']):	
		for x in xrange(0,map_grid_vars['upPltW']):
			myMap.grid[y][x]['level'] = map_prop_vars['upp_plat']
	#define upper ramp
	for y in xrange(0, map_grid_vars['upRmpH']):
		for x in xrange(map_grid_vars['upPltW'], map_grid_vars['width'] - map_grid_vars['loPltW']):
			myMap.grid[y][x]['level'] = map_prop_vars['ramp']
	#define lower platform
	for y in xrange(0, map_grid_vars['loPltH']):
		for x in xrange(map_grid_vars['width'] - map_grid_vars['loPltW'], map_grid_vars['width']):
			myMap.grid[y][x]['level'] = map_prop_vars['lwr_plat']
	#define lower ramp
	for y in xrange(map_grid_vars['loPltH'], map_grid_vars['loPltH'] + map_grid_vars['loRmpH']):
		for x in xrange(map_grid_vars['width'] - map_grid_vars['loRmpW'], map_grid_vars['width']):
			myMap.grid[y][x]['level'] = map_prop_vars['ramp']
	#define edge between upper platform and ground and long ramp and ground
	y = map_grid_vars['upPltH'] - 1; 
	for x in xrange(0, map_grid_vars['upPltW'] + map_grid_vars['upRmpW'] + 1):
		#Map[x][y].level = 9
		myMap.grid[y][x]['desc'] = map_prop_vars['edge']
	#define edge between short ramp and ground
	x = map_grid_vars['upPltW'] + map_grid_vars['upRmpW']
	for y in xrange(map_grid_vars['loPltH'], (map_grid_vars['loPltH'] + map_grid_vars['loRmpH'])):
		#Map[x][y].level = 9
		myMap.grid[y][x]['desc'] = map_prop_vars['edge']

	"""walls"""
	#define long wall along width of course (south side)
	for y in xrange(map_grid_vars['height'] - map_grid_vars['wall'], map_grid_vars['height']):
		for x in xrange(0, map_grid_vars['width']):
			myMap.grid[y][x]['desc'] = map_prop_vars['wall']
	#define short wall along width of course (north side)
	for y in xrange(map_grid_vars['upPltH'], map_grid_vars['upPltH'] + map_grid_vars['wall']):
		for x in xrange(0, map_grid_vars['upPltW'] + map_grid_vars['upRmpW']):
			myMap.grid[y][x]['desc'] = map_prop_vars['wall']
	#define long wall along height of course (west side)
	for y in xrange(map_grid_vars['upPltH'], map_grid_vars['height']):
		for x in xrange(0, map_grid_vars['wall']):
			myMap.grid[y][x]['desc'] = map_prop_vars['wall']
	#short wall along height of course (east side)
	for y in xrange(map_grid_vars['loPltH'] + map_grid_vars['loRmpH'], map_grid_vars['height']):
		for x in xrange(map_grid_vars['width'] - map_grid_vars['wall'], map_grid_vars['width']):
			myMap.grid[y][x]['desc'] = map_prop_vars['wall']

	"""start area"""
	#start area - white outHine - includes start area, next loop fixes enclosed area
	for y in xrange(map_grid_vars['height'] - map_grid_vars['wall'] - map_grid_vars['startH'] - map_grid_vars['whiteLine'], map_grid_vars['height'] - map_grid_vars['wall']):
		for x in xrange(map_grid_vars['wall'], map_grid_vars['wall'] + map_grid_vars['startW'] + map_grid_vars['whiteLine']):
			myMap.grid[y][x]['desc'] = map_prop_vars['line']
			myMap.grid[y][x]['color'] = map_prop_vars['white']
	#start area - fix enclosed area
	for y in xrange(map_grid_vars['height'] - map_grid_vars['wall'] - map_grid_vars['startH'], map_grid_vars['height'] - map_grid_vars['wall']):
		for x in xrange(map_grid_vars['wall'], map_grid_vars['wall'] + map_grid_vars['startW']):
			myMap.grid[y][x]['desc'] = map_prop_vars['start']

	"""air loading zone"""
	#air loading zone - white outline - includes enclosed space, next 2 loops fix
	for y in xrange(map_grid_vars['upPlt_2_Air'], map_grid_vars['upPltH']-map_grid_vars['upPlt_2_Air']):
		for x in xrange(0, map_grid_vars['air_long']+map_grid_vars['whiteLine']):
			myMap.grid[y][x]['desc'] = map_prop_vars['line']
			myMap.grid[y][x]['color'] = map_prop_vars['white']
	#air loading zone - fix enclosure - use two loops to account for seperating white line
	for y in xrange(map_grid_vars['upPlt_2_Air']+map_grid_vars['whiteLine'], map_grid_vars['upPlt_2_Air']+map_grid_vars['whiteLine']+map_grid_vars['zone_short']):
		for x in xrange(0,map_grid_vars['air_long']):
			myMap.grid[y][x]['desc'] = map_prop_vars['air']	
			myMap.grid[y][x]['color'] = map_prop_vars['unk']	#color unknown
	for y in xrange(map_grid_vars['upPlt_2_Air']+map_grid_vars['whiteLine']+map_grid_vars['zone_short']+map_grid_vars['whiteLine'], map_grid_vars['upPltH']-map_grid_vars['upPlt_2_Air']-map_grid_vars['whiteLine']):
		for x in xrange(0,map_grid_vars['air_long']):
			myMap.grid[y][x]['desc'] = map_prop_vars['air']	
			myMap.grid[y][x]['color'] = map_prop_vars['unk']	#color unknown

	"""cargo area"""
	#cargo storage - white outline - includes enclosed area, next loop fixes
	for y in xrange(map_grid_vars['upPltH']+map_grid_vars['wall'], map_grid_vars['upPltH']+map_grid_vars['wall']+map_grid_vars['land_long']+map_grid_vars['whiteLine']):
		for x in xrange(map_grid_vars['wall']+map_grid_vars['edge2storage'], map_grid_vars['wall']+map_grid_vars['edge2storage']+map_grid_vars['cargoL']):
			myMap.grid[y][x]['desc'] = map_prop_vars['line']
			myMap.grid[y][x]['color'] = map_prop_vars['white']
	#fix enclosed storage area and make separating white lines
	k = 0	#variable to count the nth seperating line being implemented
	m = map_grid_vars['wall'] + map_grid_vars['edge2storage']+map_grid_vars['whiteLine']	#number of tiles until the first seperating white line
	#print("width=", width, "height=", height) #debug
	while k<=13:	#there are thirteen seperating lines (14 storage slots)
		p = m + map_grid_vars['air_long']*k
		#m = m + map_grid_vars['air_long']*k
		for y in xrange(map_grid_vars['upPltH']+map_grid_vars['wall'], map_grid_vars['upPltH']+map_grid_vars['wall']+map_grid_vars['stor_long']):
			for x in xrange(p,p+map_grid_vars['zone_short']):
				#print("k=", k,"m=", m, "x=", x, "y=", y) #debug
				myMap.grid[y][x]['desc'] = map_prop_vars['storage']
				myMap.grid[y][x]['color'] = map_prop_vars['unk']  	#color unknown
		k = k + 1

	"""sea loading zone"""
	#white outline of sea loading zone
	for y in xrange(map_grid_vars['upPltH']+map_grid_vars['wall']+map_grid_vars['EdgetoSea'], map_grid_vars['upPltH']+map_grid_vars['wall']+map_grid_vars['EdgetoSea']+map_grid_vars['seaH']):
		for x in xrange(map_grid_vars['wall'], map_grid_vars['wall']+map_grid_vars['sea_long']+map_grid_vars['whiteLine']):
			myMap.grid[y][x]['desc'] = map_prop_vars['line']
			myMap.grid[y][x]['color'] = map_prop_vars['white']
	#fix enclosed sea area and make seperating white lines
	k = 0	#variable to count the ntinth seperating line being implemented
	m = map_grid_vars['upPltH'] + map_grid_vars['wall']+map_grid_vars['EdgetoSea']+map_grid_vars['whiteLine']
	while k<=5:
		p = m + map_grid_vars['air_long']*k
		#m = m + map_grid_vars['air_long']*k
		for y in xrange(p, p + map_grid_vars['zone_short']):
			for x in xrange(map_grid_vars['wall'], map_grid_vars['wall'] + map_grid_vars['sea_long']):
				myMap.grid[y][x]['desc'] = map_prop_vars['sea']
				myMap.grid[y][x]['color'] = map_prop_vars['unk']	#color unknown
		k = k + 1

	"""land loading zone"""
	#white outline of land zone
	for y in xrange(map_grid_vars['height']-map_grid_vars['wall']-map_grid_vars['land_long']+map_grid_vars['whiteLine'], map_grid_vars['height'] - map_grid_vars['wall']):
		for x in xrange(map_grid_vars['wall']+map_grid_vars['startW']+map_grid_vars['whiteLine']+map_grid_vars['start2land'],map_grid_vars['wall']+map_grid_vars['startW']+map_grid_vars['whiteLine']+map_grid_vars['start2land']+map_grid_vars['landW']):
			myMap.grid[y][x]['desc'] = map_prop_vars['line']	#marker
			myMap.grid[y][x]['color'] = map_prop_vars['white']	#white
	#fix enclosed land storage
	k = 0
	m = map_grid_vars['wall'] + map_grid_vars['startW']+map_grid_vars['whiteLine']+map_grid_vars['start2land']+map_grid_vars['whiteLine']
	while k<=5:
		p = m + map_grid_vars['air_long']*k
		#m = m+map_grid_vars['air_long']*k
		for y in xrange(map_grid_vars['height']-map_grid_vars['wall']-map_grid_vars['land_long'],map_grid_vars['height']-map_grid_vars['wall']):
			for x in xrange(p,p+map_grid_vars['zone_short']):
				myMap.grid[y][x]['desc'] = map_prop_vars['land']	#land 
				myMap.grid[y][x]['color'] = map_prop_vars['unk']	#color unknown
		k = k+1

	
	return(myMap)
